src/frontend/Sider/Behandling/Inngangsvilkår/Målgruppe/Målgruppe.tsx

```diff
@@ -1,34 +1,122 @@
+                    : periode
+            )
+        );
+    };
 
     return (
-        <Vilkårpanel tittel="Målgruppe" vilkårsresultat={vilkår.resultat}>
-            <VilkårpanelInnhold>
-                {{
-                    venstre: <MålgruppeInfo />,
-                    høyre: <VisEllerEndreVurdering vilkår={vilkår} regler={vilkårsregler.regler} />,
-                }}
-            </VilkårpanelInnhold>
-        </Vilkårpanel>
+        <Container>
+            <Heading size="medium">Målgruppe</Heading>
+            <Table>
+                <Table.Header>
+                    <Table.HeaderCell style={{ width: '20px' }} />
+                    <Table.HeaderCell>Type</Table.HeaderCell>
+                    <Table.HeaderCell>Periode</Table.HeaderCell>
+                    <Table.HeaderCell />
+                </Table.Header>
+                <Table.Body>
+                    {målgrupper.map((målgruppe) => (
+                        <Table.ExpandableRow
+                            key={målgruppe.id}
+                            togglePlacement={'right'}
+                            expansionDisabled={målgruppe.vilkår.delvilkårsett.length === 0}
+                            content={
+                                <>
+                                    <Heading size={'xsmall'}>Vilkårsvurdering</Heading>
+                                    <EndreVurderingComponent
+                                        vilkårType={målgruppe.vilkår.vilkårType}
+                                        regler={regler[målgruppe.vilkår.vilkårType].regler}
+                                        vilkår={målgruppe.vilkår}
+                                        oppdaterVilkår={oppdaterVilkår}
+                                    />
+                                </>
+                            }
+                        >
+                            <Table.DataCell width="max-content">
+                                <VilkårsresultatIkon vilkårsresultat={målgruppe.vilkår.resultat} />
+                            </Table.DataCell>
+                            <Table.DataCell>{målgruppe.type}</Table.DataCell>
+                            <Table.DataCell>
+                                {formaterIsoPeriode(målgruppe.fom, målgruppe.tom)}
+                            </Table.DataCell>
+                        </Table.ExpandableRow>
+                    ))}
+                </Table.Body>
+            </Table>
+            {skalViseLeggTilPeriode ? (
+                <LeggTilMålgruppe leggTilNyMålgruppe={leggTilNyMålgruppe} />
+            ) : (
+                <Button
+                    onClick={() => settSkalViseLeggTilPeriode((prevState) => !prevState)}
+                    size="small"
+                    style={{ maxWidth: 'fit-content' }}
+                    variant="secondary"
+                    icon={<PlusCircleIcon />}
+                >
+                    Legg til ny målgruppeperiode
+                </Button>
+            )}
```

Comment:
Jeg tolket umiddelbart `skalViseLeggTilPeriode` som en state som styrte om knappen skal vises eller ikke, men ser jo at det var feil. Kunne det være en idé å flytte knappen og staten inn i `<LeggTilMålgruppe ... />`-komponenten, og så la den selv håndtere om den er i ekspandert eller lukket modus?

Reply:
Usikker på om jeg er enig i denne, da rendrer man en del som ikke er vits å rendre med mindre man trykker på knappen, f.eks. form. Kan evt. rename denne så den gir mer mening

Reply:
Nei, man vil jo rendre akkurat det samme? Jeg tenker at knappen hører sammen med koden for å legge til, men er ikke så viktig